FiftyOne.Pipeline.CloudRequestEngine/FlowElements/CloudRequestEngine.cs

```diff
@@ -191,14 +188,32 @@ public class CloudRequestEngine :
         /// be populated after a call to the cloud service as part of 
         /// object initialization.
         /// </summary>
-        public override IEvidenceKeyFilter EvidenceKeyFilter => _evidenceKeyFilter;
+        public override IEvidenceKeyFilter EvidenceKeyFilter {
+            get
+            {
+                if (_evidenceKeyFilter == null)
+                {
+                    GetCloudEvidenceKeys();
+                }
+                return _evidenceKeyFilter;
+            }
+        }
 
         /// <summary>
         /// A collection of the properties that the cloud service can
         /// populate in the JSON response.
         /// Keyed on property name.
         /// </summary>
-        public IReadOnlyDictionary<string, ProductMetaData> PublicProperties => _publicProperties;
+        public IReadOnlyDictionary<string, ProductMetaData> PublicProperties {
+            get
+            {
+                if (_publicProperties == null)

```

Comment:
My preference would be to add a double checked lock and to modify the method to return an instance of the data structure rather than relying on the method to set the field value.

Reply:
Changes applied.